### PR TITLE
[jaeger-v2] Add support for Elasticsearch

### PR DESCRIPTION
**Pull request by hellspawn679**
#### What this PR does
added support for elasticsearch in jaeger-v2 helm chart
#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format,
will close that issue when PR gets merged)*

- fixes add support for elasticsearch for jaeger v2

#### Checklist

- [x] [DCO](https://github.com/jaegertracing/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Commits are [GPG signed](https://docs.github.com/en/github/authenticating-to-github/about-commit-signature-verification)
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (`[jaeger]` or `[jaeger-operator]`)
- [x] README.md has been updated to match version/contain new values

### CODE REVIEW

**Comment by yurishkuro** [comment]
please write complete PR title and description

File: charts/jaeger/Chart.yaml [hunk 1]
```diff
@@ -4,6 +4,8 @@ description: A Jaeger Helm chart for Kubernetes
 name: jaeger
 type: application
 version: 4.0.0
+annotations:
+  Jaegerv1Version: "1.53.0"
```
Comment (yurishkuro) [inline charts/jaeger/Chart.yaml:8]:
```suggestion
  Jaegerv1Version: "1.62.0"
```
use latest where agent was present

File: charts/jaeger/templates/collector-svc.yaml [hunk 2]
```diff
@@ -61,7 +61,7 @@ spec:
     targetPort: {{ .Values.collector.service.otlp.http.name }}
 {{- end }}
   - name: {{ .Values.collector.service.admin.name }}
```
Comment (yurishkuro) [inline charts/jaeger/templates/collector-svc.yaml:63]: What was the meaning of "admin"? In v2 there is no single admin port.

File: charts/jaeger/templates/es-user-config.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        storage:
+          traces: some_storage
+          traces_archive: another_storage
+
+
+      jaeger_storage:
+        backends:
+          some_storage:
+            elasticsearch:
+              index_prefix: {{- .Values.storage.elasticsearch.index_prefix | quote | indent 2 }}
```
Comment (yurishkuro) [inline charts/jaeger/templates/es-user-config.yaml:35]: I don't like this. `Values.storage.elasticsearch.index_prefix` is a separate domain-specific configuration language that exists only in the Operator - why do we need it? The users can provide their own custom config if they want to customize these things. If we still want to allow overriding specific fields of the built-in configuration, then those overrides should be using the OTEL config syntax, e.g. `Values.config.extensions.jaeger_storage...etc...`

File: user-config.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+service:
```
Comment (yurishkuro) [inline user-config.yaml:1]: should this file be called `user-config.yaml`? It's ES-specific.

File: charts/jaeger/templates/es-user-config.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
+{{ if (and (not .Values.userconfig) .Values.provisionDataStore.elasticsearch) }}
```
Comment (yurishkuro) [inline charts/jaeger/templates/es-user-config.yaml:1]: should this file be called "user config"? It looks like a built-in config. Also it's not even a config, but a configmap, so let's call it that.

File: user-config.yaml (hunk 4 above)
Comment (hellspawn679) [reply user-config.yaml:1]: No, if the user provides their own config, this will be used instead of `es-user-config.yaml`.

File: charts/jaeger/templates/es-user-config.yaml (hunk 3 above)
Comment (hellspawn679) [reply charts/jaeger/templates/es-user-config.yaml:35]: I wanted to keep the flag syntax same as jaeger-v1 so if the users wants to run it with old flag he can set it that way and helm will use `es-user-conifg.yaml `for that but if the user wants to provide there own config he can use the `-set-file` flag in that case we will use `user-config.yaml`

File: charts/jaeger/templates/es-user-config.yaml (hunk 5 above)
Comment (hellspawn679) [reply charts/jaeger/templates/es-user-config.yaml:1]: Ok, I will rename it to `es-configmap.yaml `

File: charts/jaeger/templates/collector-svc.yaml (hunk 2 above)
Comment (hellspawn679) [reply charts/jaeger/templates/collector-svc.yaml:63]: So what should I name it in `values.yaml`

File: .github/workflows/lint-test.yaml [hunk 6]
```diff
@@ -50,11 +50,32 @@ jobs:
       - name: Run chart-testing (install)
         run: |
           ct install --config ct.yaml \
-            --helm-extra-set-args "
+              --helm-extra-set-args "
```
Comment (pavelnikolov) [inline .github/workflows/lint-test.yaml:53]: Are the quites in the back intentional? Also should you include end of line, i.e. `\`?

File: .github/workflows/lint-test.yaml [hunk 7]
```diff
@@ -50,11 +50,32 @@ jobs:
...
+        - uses: actions/checkout@v2
+          with:
+            fetch-depth: 0
+
+        - uses: ./.github/actions/prepare-k8s
+
+        - name: Run chart-testing (install)
+          run: |
+            ct install --config ct.yaml \
+                --helm-extra-set-args "
```
Comment (pavelnikolov) [inline .github/workflows/lint-test.yaml:72]: Again, end of line `\` missing.

File: charts/jaeger/templates/collector-svc.yaml (hunk 2 above)
Comment (yurishkuro) [reply charts/jaeger/templates/collector-svc.yaml:63]: This specific port is healthcheck

File: charts/jaeger/templates/es-configmap.yaml [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        storage:
+          traces: some_storage
+          traces_archive: another_storage
+
+
+      jaeger_storage:
+        backends:
+          some_storage:
+            elasticsearch:
+              index_prefix: {{- .Values.config.extensions.jaeger_storage.backends.elasticsearch.index_prefix | quote | indent 2 }}
```
Comment (yurishkuro) [inline charts/jaeger/templates/es-configmap.yaml:35]: why do you need to use `.Values.config.extensions.jaeger_storage.backends.elasticsearch. index_prefix ` reference here?

File: charts/jaeger/values.yaml [hunk 9]
```diff
@@ -32,6 +32,12 @@ config:
 
     jaeger_storage:
       backends:
+        elasticsearch:
```
Comment (yurishkuro) [inline charts/jaeger/values.yaml:35]: this is not a valid syntax. It should be 
```
backends:
  {some name}:
    elasticsearch: // this refers to the type of storage
```

File: user-config.yaml (hunk 4 above)
Comment (yurishkuro) [reply user-config.yaml:1]: is this file meant to be an example config?

**Review by yurishkuro — COMMENTED** [review]
please document in the README how the user is expected to use the chart with ES, using different options such as providing params via `--set` or a full config via `--set-file`

File: .github/workflows/lint-test.yaml [hunk 10]
```diff
@@ -50,11 +50,32 @@ jobs:
...
+          with:
+            fetch-depth: 0
+
+        - uses: ./.github/actions/prepare-k8s
+
+        - name: Run chart-testing (install)
+          run: |
+            ct install --config ct.yaml \
+                --helm-extra-set-args " \
+                --set provisionDataStore.cassandra=false \
```
Comment (yurishkuro) [inline .github/workflows/lint-test.yaml:73]: this should be default, can we avoid setting it?

File: .github/workflows/lint-test.yaml [hunk 11]
```diff
@@ -50,11 +50,32 @@ jobs:
       - name: Run chart-testing (install)
         run: |
           ct install --config ct.yaml \
-            --helm-extra-set-args "
+              --helm-extra-set-args "  \
               --set provisionDataStore.cassandra=false \
```
Comment (yurishkuro) [inline .github/workflows/lint-test.yaml:54]: is this flag relevant to memory storage?

File: charts/jaeger/README-v2.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+
+### 1. Jaeger All-in-One Mode
+
+Jaeger’s all-in-one mode combines the Jaeger Agent, Collector, and Query into a single pod for simplicity. Use this mode for testing or lightweight deployments.
+
+#### **Command to Run All-in-One**
+
+```bash
+helm install <chart_name> charts/jaeger \
+    --set provisionDataStore.cassandra=false \
```
Comment (yurishkuro) [inline charts/jaeger/README-v2.md:19]: is it releavant?

File: .github/workflows/lint-test.yaml [hunk 13]
```diff
@@ -50,11 +50,32 @@ jobs:
...
               --set storage.type=memory \
               --set allInOne.enabled=true \
               --set agent.enabled=false \
               --set collector.enabled=false \
               --set query.enabled=false "
```
Comment (yurishkuro) [inline .github/workflows/lint-test.yaml:59]: I think these should automatically be treated as false if `allInOne` is true

File: charts/jaeger/templates/es-configmap.yaml [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+      jaeger_query:
+        storage:
+          traces: primary_store
+          traces_archive: archive_store
+
+
+      jaeger_storage:
+        backends:
+          primary_store:
+            elasticsearch:
```
Comment (yurishkuro) [inline charts/jaeger/templates/es-configmap.yaml:34]: I don't mind merging like this as a milestone, but I think this file should not be specific to ES only, it can use the flags indicating which storage type the user selected and creating the corresponding configs for that storage. Otherwise we would have to duplicate this overall config for each storage type.

File: charts/jaeger/templates/es-configmap.yaml [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+              username: {{- .Values.config.extensions.jaeger_storage.backends.archive_store.elasticsearch.user | quote | indent 2 }}
+              {{- if .Values.config.extensions.jaeger_storage.backends.archive_store.elasticsearch.password }}
+              password: {{- .Values.config.extensions.jaeger_storage.backends.archive_store.elasticsearch.password | quote | indent 2 }}
+              {{- end }}
+
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+          http:
```
Comment (yurishkuro) [inline charts/jaeger/templates/es-configmap.yaml:54]: we need to set endpoints to 0.0.0.0:portNum, see https://github.com/jaegertracing/jaeger/issues/6209

File: charts/jaeger/values.yaml [hunk 16]
```diff
@@ -33,9 +33,19 @@ config:
     jaeger_storage:
```
Comment (yurishkuro) [inline charts/jaeger/values.yaml:33]: help me understand - this `config` structure is not actually used as a config, right? It only provides the shape for the properties that looks like OTEL config?

File: charts/jaeger/values.yaml [hunk 17]
```diff
@@ -566,7 +576,7 @@ collector:
         # name: otlp-http
         # port: 4318
         # nodePort:
-    admin:
+    healthCheck:
       name: admin
```
Comment (yurishkuro) [inline charts/jaeger/values.yaml:580]: do we need to change the name away from `admin`? It's an inconsistency

File: user-config.yaml (hunk 4 above)
Comment (yurishkuro) [reply user-config.yaml:1]: ?

File: charts/jaeger/templates/user-config.yaml [hunk 18]
```diff
@@ -1,3 +1,4 @@
+# this is a user-provided config file that will be mounted into the all-in-one deployments
```
Comment (yurishkuro) [inline charts/jaeger/templates/user-config.yaml:1]:
```suggestion
# Generates a config map from a file provided by user via `--set-file userconfig=`
```

File: charts/jaeger/templates/query-svc.yaml [hunk 19]
```diff
@@ -24,7 +24,7 @@ spec:
     protocol: TCP
     targetPort: grpc
   - name: {{ .Values.query.service.admin.name }}
```
Comment (yurishkuro) [inline charts/jaeger/templates/query-svc.yaml:26]:
```suggestion
  - name: {{ .Values.query.service.healthcheck.name }}
```

File: charts/jaeger/templates/agent-ds.yaml [hunk 20]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.agent.enabled -}}
+{{- if and .Values.agent.enabled (not .Values.allInOne.enabled)  -}}
```
Comment (yurishkuro) [inline charts/jaeger/templates/agent-ds.yaml:1]: > `.Values.allInOne.enabled`

I don't think you should have made this change. I simply meant to change the defaults in values.yaml, not to introduce these massive changes across the chart.

File: charts/jaeger/values.yaml [hunk 21]
```diff
@@ -33,17 +33,27 @@ config:
...
           memory:
             max_traces: 100000
 
   receivers:
     otlp:
       protocols:
-        grpc:
-        http:
+        grpc: 0.0.0.0:4317
+        http: 0.0.0.0:4318
```
Comment (yurishkuro) [inline charts/jaeger/values.yaml:56]:
```suggestion
        grpc:
          endpoint: 0.0.0.0:4317
        http:
          endpoint: 0.0.0.0:4318
```

File: charts/jaeger/templates/collector-deploy.yaml [hunk 22]
```diff
@@ -123,7 +126,7 @@ spec:
         - containerPort: {{ .Values.collector.service.http.port }}
           name: http
           protocol: TCP
-        - containerPort: 14269
+        - containerPort: 13133
           name: admin
```
Comment (yurishkuro) [inline charts/jaeger/templates/collector-deploy.yaml:130]: should this not be called `healthcheck`? Is this name used anywhere?

**Review by yurishkuro — APPROVED** [review]
